### ISSUE TITLE
Poor support for Korean

### ISSUE THREAD

**Issue by chengxilo** [LibreTranslate/LibreTranslate#828]
First of all, thanks for the hard work to develop LibreTranslate, it's a great work to build such a powerful project.

Recently I am trying to build some translate service base on LibreTranslate.
And here is a problem in found during my development:

<img width="1370" height="1015" alt="Image" src="https://github.com/user-attachments/assets/c5d985fa-1e25-4125-b39b-5840500fcfe9" />

<img width="1381" height="1022" alt="Image" src="https://github.com/user-attachments/assets/3b1b4107-9f3a-4dea-9004-d1bfdfb4a811" />

When translating single word, the result seems to have low confidence (with a wrong answer)
Would LibreTranslate get further enhancement in its quility for single word?

**Comment by PJ-Finlay** [comment]
This is a known issue with Argos Translate where it struggles to translate individual words. Argos Translate is much better at translating full sentences for some reason. I've tried including Wiktionary data in the training dataset but still get mediocre results. 

The Korean-English models are also worse than a lot of the other Argos Translate models. There's probably room for improvement if you want to train a new en_ko model.

**Comment by pierotofy** [comment]
Yeah it's an interesting issue that shows up often in the transformer architecture; single word inputs are often difficult for the model to translate. I actually thought about an interesting workaround that works by "seeding" the input with a longer sequence via a known translation using a two step process.

1. Translate a sentence such as `The person said: %1` --> `사람이 말했다: %1`
2. Translate another sentence with the token now replacing %1: `The person said: eat` --> `사람이 말했다 : 먹고`. Do a regex subgroup match on the result of the second result with the first.
3. If the regex is a match, return the regex match subgroup. Otherwise try to translate the single word like we do currently and hope for the best.

**Comment by AHVSSATHVIK** [comment]
Thanks for your amazing work on LibreTranslate — it’s a fantastic project!

I’ve been building a translation service based on LibreTranslate and noticed that
**single-word translations sometimes give incorrect or low-confidence results**.
After testing, it seems this happens because the underlying models (Argos Translate)
are primarily trained on sentence-level parallel data rather than isolated words.

### Possible Causes
- Lack of context: neural models are optimized for full sentences, not standalone words.
- Ambiguity: without sentence context, the model can’t choose the right meaning (e.g., “bank” = riverbank or financial).
- Limited dictionary-style coverage in training corpora.

### Proposed Solutions / Enhancements
1. **Add or fine-tune a model with dictionary-level data**  
   Include parallel wordlists or bilingual lexicons during training to improve isolated word performance.
2. **Add a fallback “dictionary lookup” layer**  
   When only a single token (no spaces) is submitted, the service could:
   - Check a bilingual dictionary or glossary first,
   - Fall back to the translation model if not found.
3. **Context hint workaround**  
   Temporarily, developers can wrap single words in a minimal sentence
   (e.g., “Translate the word ‘bank’.”) to improve model accuracy.
4. **Optional API flag**  
   A flag like `"mode": "word"` could be added to trigger dictionary-based or context-enriched handling.

### Example Implementation Idea
For now, I’ve implemented a small pre-processing step in my service:
```js
if (q.split(/\s+/).length === 1) {
  // Handle single word separately
  result = dictionaryLookup(q) || libreTranslate(q);
}

**Comment by Akshat-Rai0** [comment]
After doing some more testing, it looks like this is probably not something we can reliably fix on the LibreTranslate side.

I also tried adding some extra context around the input (context injection) to help the model figure out the intended meaning, but the results are still not consistent enough.

At this point, it seems more like an inherent limitation of the underlying translation model when dealing with isolated or ambiguous words. Even with additional context, the model can still produce an unexpected translation, so there doesn’t seem to be a reliable fix we can implement in LibreTranslate itself.

I think we can close this issue as a model-level limitation, since fixing it properly would likely require changes to the underlying model or its training data rather than LibreTranslate.

**Comment by DimmoK** [comment]
Hello! I use this for my translate page
```
curl -X POST http://127.0.0.1:5000/translate   -H "Content-Type: application/json"   -d '{
    "q":"дождь",
    "source":"ru",
    "target":"en",
    "alternatives":5
  }'
{"alternatives":["rainfall","rainbow","raindrop","rainstorm","raining"],"translatedText":"rain"}

curl -X POST http://127.0.0.1:5000/translate   -H "Content-Type: application/json"   -d '{
    "q":"платить",
    "source":"ru",
    "target":"en",
    "alternatives":5
  }'
{"alternatives":["pay","pay up","charge","pay back","payback"],"translatedText":"pay off"}
```

How it is looking in web
<details>
<summary>IMAGES</summary>

<img width="1233" height="732" alt="Image" src="https://github.com/user-attachments/assets/c1a0e98c-b6a2-447e-885f-7294e0b32c89" />

<img width="1233" height="732" alt="Image" src="https://github.com/user-attachments/assets/664c33e6-5613-4edf-b34d-6330c58a6c9a" />

</details>
